api/test/audio_integration_test.py

```diff
@@ -27,11 +27,29 @@
 
 import pytest
 import requests
+from django_redis import get_redis_connection
+
+from catalog.api.utils.validate_images import CACHE_PREFIX
+
+
+@pytest.fixture
+def force_first_page_audio_validity():
+    first_page = requests.get(
+        f"{API_URL}/v1/audio?filter_dead=False", verify=False
+    ).json()
+    statuses = {f"{CACHE_PREFIX}{item['url']}": 200 for item in first_page["results"]}
+    with get_redis_connection() as redis:
+        redis.mset(statuses)
+
+    yield
+
+    with get_redis_connection() as redis:
+        redis.delete(*list(statuses.keys()))
 
 
 @pytest.fixture
-def audio_fixture():
-    res = requests.get(f"{API_URL}/v1/audio?q=", verify=False)
+def audio_fixture(force_first_page_audio_validity):
```

Comment:
I'm looking into this thumbnail issue a bit more - the invocation of this fixture indeed causes the thumbnails to be `null`. I switched from `main` to this branch and queried the API a few times and saw that the thumbnails still appeared. As soon as I ran `j api-test -k test_audio_thumb` the thumbnails all became `null`.

Reply:
Okay now I'm not so sure 🙈 I ran `flushall` on the redis cache and the thumbnails are still `null`. I'll continue to look.

Reply:
The link validation cache _shouldn't_ have anything to do with it.

Curiously, this problem doesn't appear to affect the CI test run as it's passing fine here :thinking:

Reply:
Also, if you look at the sample data, the thumbnails are indeed set to null for all audio, so it's not a mistake there. The curious thing is why it ever passed (and why is it passing in CI? :woozy_face:)

Reply:
Okay I'm seeing the following behavior on `main` after I run `just recreate`:
1. First hit to http://localhost:8000/v1/audio/ - no thumbnails
2. Second hit to http://localhost:8000/v1/audio/ - no _results_
3. Third hit to http://localhost:8000/v1/audio/ - results again with thumbnails now

I've been able to reproduce this a few times now. Judging from the logs, it looks like it's clearing out the Wikimedia results (which might not have thumbnails?), and the end result is only Jamendo results.

Reply:
Oh dear, it actually looks like `validate_images` is improperly named:
```
openverse-api-web-1  | [2022-08-15 22:26:20,217 - catalog.api.utils.validate_images.validate_images - 105][WARNING] [7d18c2d30cc64f309ed1dbc4cc5db9b4] Image validation failed due to rate limiting or blocking. url=https://upload.wikimedia.org/wikipedia/commons/8/88/LL-Q1860_%28eng%29-Vealhurl-walker.wav status=403 
```

It seems that it should probably be `validate_media`, I'll make an issue for this.

Reply:
Okay this feels really inadvisable, but with this diff I was able to get all the tests to pass:

```diff
diff --git a/api/test/audio_integration_test.py b/api/test/audio_integration_test.py
index c732ceb5..248e6cc9 100644
--- a/api/test/audio_integration_test.py
+++ b/api/test/audio_integration_test.py
@@ -35,7 +35,7 @@ from catalog.api.utils.validate_images import CACHE_PREFIX
 @pytest.fixture
 def force_first_page_audio_validity():
     first_page = requests.get(
-        f"{API_URL}/v1/audio?filter_dead=False", verify=False
+        f"{API_URL}/v1/audio?filter_dead=False&source=jamendo", verify=False
     ).json()
     statuses = {f"{CACHE_PREFIX}{item['url']}": 200 for item in first_page["results"]}
     with get_redis_connection() as redis:
@@ -49,7 +49,7 @@ def force_first_page_audio_validity():
 
 @pytest.fixture
 def audio_fixture(force_first_page_audio_validity):
-    res = requests.get(f"{API_URL}/v1/audio", verify=False)
+    res = requests.get(f"{API_URL}/v1/audio?source=jamendo", verify=False)
     assert res.status_code == 200
     parsed = json.loads(res.text)
     return parsed
@@ -60,7 +60,7 @@ def test_search(audio_fixture):
 
 
 def test_search_category_filtering(audio_fixture):
-    search_by_category("audio", "music", audio_fixture)
+    search_by_category("audio", "news", audio_fixture)
     search_by_category("audio", "pronunciation", audio_fixture)
```

I have to wonder if the reason it's passing on CI is because Wikimedia isn't rate limiting whatever machine the tests are running on, but it _is_ rate limiting all of us 😮

Reply:
We keep getting 403s from Wikimedia - that's supposed to be addressed by #877, perhaps that might be necessary in order to merge this too?

Reply:
I tried adding this commit onto #877 and I'm still getting errors 😞 I think we may just need to be specific about which results we're querying (e.g. `source=jamendo`), at least until we're able to mock all of these requests. As far as I can tell, this all comes back to the dead link filtering. What do you think?

Reply:
I fine with just filtering for a specific source for these tests in particular. I don't think these tests are set up in any way that makes sense to me. I'd rather use fixture/factory data that we actually encode into the tests themselves instead of sample data that is (a) difficult to maintain and (b) difficult to have insights into.